sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -53,9 +53,11 @@ async def on_request(self, request: PipelineRequest) -> None:
                 scope = challenge.get_scope() or challenge.get_resource() + "/.default"
                 # Exclude tenant for AD FS authentication
                 if challenge.tenant_id and challenge.tenant_id.lower().endswith("adfs"):
-                    self._token = await self._credential.get_token(scope)
+                    self._token = await self._credential.get_token(scope, claims=challenge.claims)
```

Comment:
You also need to explicitly enable CAE every time you call `get_token`:
```suggestion
                    self._token = await self._credential.get_token(scope, claims=challenge.claims, enable_cae=True)
```
Otherwise, depending on tenant configuration, you get an error or conditional access policies don't apply to the token i.e. the tenant can't revoke it